geolocator_platform_interface/lib/src/implementations/method_channel_geolocator.dart

```diff
@@ -21,13 +21,19 @@ class MethodChannelGeolocator extends GeolocatorPlatform {
   static const _eventChannel =
       EventChannel('flutter.baseflow.com/geolocator_updates');
 
+  /// The event channel used to receive [LocationServiceStatus] updates from the
+  /// native platform.
+  static const _serviceStatusEventChannel =
+      EventChannel('flutter.baseflow.com/geolocator_service_updates');
+
   /// On Android devices you can set [forceAndroidLocationManager]
   /// to true to force the plugin to use the [LocationManager] to determine the
   /// position instead of the [FusedLocationProviderClient]. On iOS this is
   /// ignored.
   bool forceAndroidLocationManager = false;
 
   Stream<Position>? _positionStream;
+  Stream<bool>? _serviceStatusStream;
```

Comment:
It would be better if we could return an enum value explaining the state instead of a boolean. So let's define a `ServiceStatus` enum that looks something like this:

```dart
/// Describes the current state of the location service on the native platform.
enum ServiceStatus {
  /// Indicates that the location service on the native platform is enabled.
  enabled,
  /// Indicates that the location service on the native platform is enabled.
  disabled,
}
```

And make sure this is wat we return on the stream:
```suggestion
  Stream<ServiceStatus>? _serviceStatusStream;
```

This makes the code self-explanatory and makes it easier for developers to understand what a certain result means.